retrofit/src/test/java/retrofit2/CallTest.java

```diff
@@ -978,4 +985,66 @@
     }
     assertThat(writeCount.get()).isEqualTo(1);
   }
+
+  @Test public void queryParamsSkippedIfConvertedToNull() throws IOException, InterruptedException {
```

Comment:
Yeah, I wasn't sure where to put them. Open to suggestions and I'll make that change